src/combat.cpp

```diff
@@ -905,6 +905,10 @@ void Combat::doAreaCombat(Creature* caster, const Position& position, const Area
 						if (g_game.combatBlockHit(damageCopy, caster, creature, params.blockedByShield, params.blockedByArmor, params.itemId != 0)) {
 							continue;
 						}
+						if (casterPlayer && !damage.leeched) {
```

Comment:
same, why move there?
I agree with that bool though.

Reply:
You're completely right, should check for highest damage and then leech.

Reply:
> For area spells this is actually still wrong. Cip calculates the highest damage from all the creatures that were hit in area spell and then calculates leech amount out of it. Although it should fix leeching for every target.

This has been altered, it now stores the highest damage done in areaattack and leeches off that.